### DISCUSSION TITLE
How to set the isolation level using RDF4J's REST API

### DISCUSSION THREAD

**Discussion by arrascue** [eclipse-rdf4j/rdf4j#3876] [Q&A] [answered]
Hello,
I have a repository (Native Store) running on an rdf4j-server (v3.7.4). 
The REST API documentation states that to start a transaction, one can send a POST request to /rdf4j-server/repositories/<ID>/transactions and that optionally, one can set the desired isolation-level.
To me it is not clear how to set up that isolation-level. 

If I try to add the isolation as a request parameter
```
params = {'isolation-level': 'READ_COMMITTED'} 
response = requests.post(transaction_resource, params=params)
```
I get a response with 500 status

If I try to add it in the header:
```
headers = {'isolation-level' : 'READ_COMMITTED'}
response = requests.post(transaction_resource, headers=headers)
```
It returns a transaction ID on a 201 response, but I think it is simply ignoring the isolation level, since I can write anything on it, like 'isolation-level' : 'blabla' and it would still work.

If I access the repository programatically with an HTTPRepository I get 'null' when calling `conn.getIsolationLevel()`. 
I can set the isolation level programatically with `conn.setIsolationLevel(IsolationLevels.READ_COMMITTED);` but this seems to apply it only to the connection. If I access it from a different connection it would return a null isolation level.

Does anyone knows how to set up correctly the isolation-level on the request or to verify if the transaction id will behave as specified in the isolation-level?

Thank you very much

**Comment by hmottestad** [comment, marked as answer]
Can you try with this param instead?

`transaction-setting__isolation-level`

**Reply by arrascue** [reply]
Thank you!
It works with that parameter but I have noticed that it accepts any value. 
Is there a way to know if the isolation-level is actually supported and used for the transaction?

**Reply by hmottestad** [reply]
Maybe you can enable logging on the server side?

**Comment by nguyenm100** [comment]
isolation-level=XXX should be fixed in 4.1.3 (transaction-setting__xxx is still broken).  Can you try and confirm?
